README.md
# Simulado
Programa em c++ para realizar simulados via terminal

## Compilar no Linux
    g++ simulado.cpp -o simulado

## Executar no Linux
    ./simulado

changelog.txt
24/06/2017
  Separacao das funcoes em arquivos de cabecalhos
26/10/2006
	Analise do projeto
02/11/2006
	Inicio do desenvolvimento
	Menu principal
	Realiza Simulados
03/11/2006
	Criacao das funcoes auxiliares
	Aperfeicoamento da rotina realiza simulados
03/11/2006
	Correcoes de bugs
06/11/2006
	Solucao definitiva para leitura somente nos arquivos para achar as perguntas e respostas
08/11/2006
	Aperfeicoamento das funcoes de cores
09/11/2006
	Correcoes para converter inteiros para caractere nas opcoes de escolha
	Correcao para avisar o usuario quando digitar uma resposta invalida e permanecer na pergunta
11/11/2006
	Opcao para o usuario abandonar o simulado a qualquer momento, mostrando o
	resultado.
	Rotina para exibir informacoes sobre o sistema.
	Gravacao do resultado em arquivo.
	Digitacao do nome do usuario na entrada do sistema.
13/11/2006
	Gravacao da data no arquivo resultados.dat
14/11/2006
	Criacao da rotina para criacao do arquivo ranking.dat para selecionar quem mais acertou e errou.
15/11/2006
	Ajustes nos calculos de media na tela de resultados.
16/11/2006
	Correcao do erro ao digitar um mes que nao tem resultados, dava erro ao dividir as medias por zero

funcoes.h
#include "constantes.h"

// Procedimento para mudar a cor da fonte
// Parametros: cor da letra, cor de fundo
void corFonte( const char * const corLetra, const char * const corFundo ) {
  int corL = 0, corF = 0;

  if ( corLetra == "preto" ) corL = 30;
  if ( corLetra == "vermelho" ) corL = 31;
  if ( corLetra == "verde" ) corL = 32;
  if ( corLetra == "amarelo" ) corL = 33;
  if ( corLetra == "azul" ) corL = 34;
  if ( corLetra == "rosa" ) corL = 35;
  if ( corLetra == "ciano" ) corL = 36;
  if ( corLetra == "branco" ) corL = 37;

  if ( corFundo == "preto" ) corF = 40;
  if ( corFundo == "vermelho" ) corF = 41;
  if ( corFundo == "verde" ) corF = 42;
  if ( corFundo == "amarelo" ) corF = 43;
  if ( corFundo == "azul" ) corF = 44;
  if ( corFundo == "rosa" ) corF = 45;
  if ( corFundo == "ciano" ) corF = 46;
  if ( corFundo == "branco" ) corF = 47;

  if ( SO == 'x' ) // Se o sistema operacional for Unix/Linux
    cout << "\e[" << corF << ";" << corL << ";1m"; // Usa os comandos do shell
}

// Procedimento para limpar a tela
// Parametros: nenhum
void limpaTela( void ) {
  if ( SO == 'x' ) // Se o sistema operacional for Unix/Linux
    cout << "\e[H\e[2J"; // Usa os comandos do shell
  else // Se o sistema operacional for Windows
    system( "CLS" );
}

// Procedimento para dar pausa na tela
// Parametros: nenhum
void pausa( void ) {
  if ( SO == 'x' ) { // Se o sistema operacional for Unix/Linux
    corFonte( "verde", COR_FUNDO );
    cout << "\nTecle < ENTER > para continuar...\n";
    corFonte( "branco", COR_FUNDO );
    system( "read -r REPLY" );
  } else { // Se o sistema operacional for Windows
    cout << endl;
    system( "PAUSE" );
  }
}

// Procedimento para replicar um caracter n vezes
// Parametros: caractere a ser impresso, numero de vezes para ser impresso
void replica( char caractere, int nvezes ) {
  for ( int i = 1; i <= nvezes; i++ )
    cout << caractere;

  cout << endl;
}

// Procedimento para exibir mensagem na tela
// Parametros: texto da mensagem
void mensagem( const char * const aviso ) {
  cout << endl;
  corFonte( "branco", "vermelho" );
  replica( '*', strlen( aviso ) + 8 );
  cout << "*** " << aviso << " ***\n";
  replica( '*', strlen( aviso ) + 8 );
  corFonte( "branco", COR_FUNDO );
  pausa();
}

// Procedimento para exibir um titulo destacado na tela
// Parametros: texto do titulo, caractere para as linhas, tamanho da linha
void titulo( const char * const texto, char caractere, int tamanhoLinha ) {
  corFonte( "branco", COR_FUNDO );
  limpaTela();
  replica( caractere, tamanhoLinha );
  corFonte( "amarelo", COR_FUNDO );
  cout << texto << "\n";
  corFonte( "branco", COR_FUNDO );
  replica( caractere, tamanhoLinha );
  corFonte( "branco", COR_FUNDO );
}

// Procedimento para exibir um texto colorido
// Parametros: texto, cor da letra
void textoColorido( const char * const texto, const char * const cor ) {
  corFonte( cor, COR_FUNDO );
  cout << texto;
  corFonte( "branco", COR_FUNDO );
}


arquivos.h
#include <fstream>
#include "constantes.h"

// Rotina para criar os arquivos que nao existirem
void criaArquivos( void ) {
  // Cria o arquivo simulados.dat
  // Estrutura: codigo do simulado, descricao do simulado
  ifstream inSimulados( "simulados.dat", ios::in );

  if ( !inSimulados ) { // Se o arquivo nao existir, deve cria-lo
    ofstream outSimulados( "simulados.dat", ios::out );

    outSimulados << "1 Simulado de Windows\n";
    outSimulados << "2 Simulado de Linux\n";

    outSimulados.close();
  }
  inSimulados.close();

  // Cria o arquivo perguntas.dat
  // Estrutura: codigo do simulado, codigo da pergunta, descricao da pergunta
  ifstream inPerguntas( "perguntas.dat", ios::in );

  if ( !inPerguntas ) { // Se o arquivo nao existir, deve cria-lo
    ofstream outPerguntas( "perguntas.dat", ios::out );

    outPerguntas << "1 1 O que significa Windows?\n";
    outPerguntas << "1 2 Qual empresa e proprietaria do Windows?\n";
    outPerguntas << "1 3 Qual tecla de atalho fecha uma janela no Windows?\n";
    outPerguntas << "1 4 Qual e o pacote de escritorios do Windows?\n";
    outPerguntas << "1 5 Qual e a ultima versao do Windows?\n";

    outPerguntas << "2 1 O que e o Linux?\n";
    outPerguntas << "2 2 Quem criou o Linux?\n";
    outPerguntas << "2 3 Em quantos tipos de computador o Linux e usado?\n";
    outPerguntas << "2 4 Qual e o gerenciador de janelas mais usado no Linux?\n";
    outPerguntas << "2 5 O que significa LAMP?\n";
    outPerguntas << "2 6 Qual das distribuicoes Linux abaixo e a mais antiga?\n";

    outPerguntas.close();
  }
  inPerguntas.close();

  // Cria o arquivo respostas.dat
  // Estrutura: codigo do simulado, codigo da pergunta, codigo da resposta,
  // resposta correta (s/n), descricao da resposta
  ifstream inRespostas( "respostas.dat", ios::in );

  if ( !inRespostas ) { // Se o arquivo nao existir, deve cria-lo
    ofstream outRespostas( "respostas.dat", ios::out );

    // Respostas de Windows
    outRespostas << "1 1 1 s Janelas\n";
    outRespostas << "1 1 2 n Portas\n";
    outRespostas << "1 1 3 n Telhados\n";
    outRespostas << "1 1 4 n Quartos\n";

    outRespostas << "1 2 1 n IBM\n";
    outRespostas << "1 2 2 s Microsoft\n";
    outRespostas << "1 2 3 n HP\n";

    outRespostas << "1 3 1 n Shift+F4\n";
    outRespostas << "1 3 2 n Ctrl+F4\n";
    outRespostas << "1 3 3 s Alt+F4\n";
    outRespostas << "1 3 4 n Ctrl+Alt+Del\n";

    outRespostas << "1 4 1 n OpenOffice\n";
    outRespostas << "1 4 2 s MS Office\n";
    outRespostas << "1 4 3 n StarOffice\n";

    outRespostas << "1 5 1 s Vista\n";
    outRespostas << "1 5 2 n 2003\n";
    outRespostas << "1 5 3 n XP\n";

    // Respostas de Linux
    outRespostas << "2 1 1 n Um sistema de arquivos\n";
    outRespostas << "2 1 2 s Um sistema operacional\n";
    outRespostas << "2 1 3 n Um sistema de automacao\n";
    outRespostas << "2 1 4 n Um sistema de redes\n";

    outRespostas << "2 2 1 n Richard Stallman\n";
    outRespostas << "2 2 2 n Andrew Tanembaum\n";
    outRespostas << "2 2 3 s Linus Torvalds\n";

    outRespostas << "2 3 1 n Em pc's\n";
    outRespostas << "2 3 2 n Em Mainframes\n";
    outRespostas << "2 3 3 s Na maioria das arquiteturas\n";
    outRespostas << "2 3 4 n Em palmtops\n";

    outRespostas << "2 4 1 n GNOME\n";
    outRespostas << "2 4 2 s KDE\n";
    outRespostas << "2 4 3 n WindowMaker\n";
    outRespostas << "2 4 4 n Fluxbox\n";

    outRespostas << "2 5 1 n Sistema de arquivos lampada\n";
    outRespostas << "2 5 2 s Linux/Apache/Mysql/Php\n";
    outRespostas << "2 5 3 n Linux Advanced Media Player\n";

    outRespostas << "2 6 1 n RedHat\n";
    outRespostas << "2 6 2 n Debian\n";
    outRespostas << "2 6 3 s Slackware\n";
    outRespostas << "2 6 4 n SuSE\n";
    outRespostas << "2 6 5 n Mandriva\n";
    outRespostas << "2 6 6 n Gentoo\n";

    outRespostas.close();
  }
  inRespostas.close();
}


resultados.h
#include "constantes.h"

// Procedimento para exibir os resultados dos simulados
void resultados( void ) {
  int numeroMes, dia, mes, ano, simulado, acertos, erros, questoes;
  string usuario;

  int totalSimulados = 0;
  int totalAcertos = 0;
  int totalErros = 0;
  int totalQuestoes = 0;

  string rUsuario;
  int rAcertos, rErros;
  char achouRankingUsuario;

  int maiorAcerto = 0;
  int maiorErro = 0;
  string usuarioMaisAcertou;
  string usuarioMaisErrou;

  titulo( "Resultados", '=', REPETE + 10 );
  textoColorido( "\nDigite o n. do mes da pesquisa: ", "amarelo" );
  corFonte( "verde", COR_FUNDO );
  cin >> numeroMes;
  corFonte( "branco", COR_FUNDO );

  // Limpa o arquivo de ranking
  ofstream delRanking( "ranking.dat", ios::trunc );
  delRanking.close();

  // Abre o arquivo de resultados para leitura
  ifstream inResultados( "resultados.dat", ios::in );

  if ( inResultados ) { // Se conseguiu abri-lo
    inResultados >> dia >> mes >> ano >> usuario >> simulado >> acertos >> erros >> questoes;

    while ( !inResultados.eof() ) { // Executa ate chegar ao final do arquivo
      if ( numeroMes == mes ) { // Se for do mes escolhido
        totalSimulados++;
        totalAcertos += acertos;
        totalErros += erros;
        totalQuestoes += questoes;

        achouRankingUsuario = 'n';

        // Abre o arquivo de ranking para leitura
        ifstream inRanking( "ranking.dat" );

        if ( inRanking ) { // Se conseguiu abri-lo
          inRanking >> rUsuario >> rAcertos >> rErros; // Le a primeira linha do arquivo

          while ( !inRanking.eof() ) { // Executa enquanto nao chegar no final dele
            // Se encontrar o usuario no arquivo de ranking, na grava-lo mais no ranking
            if ( usuario == rUsuario ) {
              achouRankingUsuario = 's';
              break;
            }

            inRanking >> rUsuario >> rAcertos >> rErros; // Le a proxima linha do arquivo
          }
        }
        inRanking.close();

        // Se nao achou o usuario no arquivo de ranking, deve grava-lo
        if ( achouRankingUsuario == 'n' ) {
          rUsuario = usuario;
          rAcertos = 0;
          rErros = 0;

          // Abre o arquivo de resultados para leitura de acertos e erros do usuario
          ifstream inTotais( "resultados.dat" );

          if ( inTotais ) { // Se conseguiu abri-lo
            inTotais >> dia >> mes >> ano >> usuario >> simulado >> acertos >> erros >> questoes;

            while ( !inTotais.eof() ) {
              // Soma dos acertos e erros do usuario e mes selecionado
              if ( usuario == rUsuario && numeroMes == mes ) {
                rAcertos += acertos;
                rErros += erros;
              }

              inTotais >> dia >> mes >> ano >> usuario >> simulado >> acertos >> erros >> questoes;
            }
          }
          inTotais.close();

          // Abre o arquivo de ranking para gravacao
          ofstream outRanking ( "ranking.dat" , ios::app );
          outRanking << rUsuario << ' ' << rAcertos << ' ' << rErros << '\n';
          outRanking.close();
        }
      }

      inResultados >> dia >> mes >> ano >> usuario >> simulado >> acertos >> erros >> questoes;
    }
    inResultados.close();

    // Abre o arquivo de ranking para leitura dos melhores e piores desempenhos
    ifstream inRanking( "ranking.dat" );

    if ( inRanking ) { // Se conseguiu abri-lo
      inRanking >> rUsuario >> rAcertos >> rErros; // Le a primeira linha do arquivo

      while ( !inRanking.eof() ) { // Executa enquanto nao chegar no final dele
        if ( rAcertos > maiorAcerto ) {
          maiorAcerto = rAcertos;
          usuarioMaisAcertou = rUsuario;
        }

        if ( rErros > maiorErro ) {
          maiorErro = rErros;
          usuarioMaisErrou = rUsuario;
        }

        inRanking >> rUsuario >> rAcertos >> rErros; // Le a proxima linha do arquivo
      }
    }
    inRanking.close();
  }

  if ( totalSimulados > 0 ) {
    textoColorido( "\nSimulados realizados....: ", "amarelo" );
    cout << totalSimulados << endl;
    textoColorido( "\nTotal/Media de Acertos..: ", "amarelo" );
    cout << totalAcertos << " / " << totalAcertos/totalSimulados << endl;
    textoColorido( "\nTotal/Media de Erros....: ", "amarelo" );
    cout << totalErros << " / " << totalErros/totalSimulados << endl;
    textoColorido( "\nUsuario que mais acertou: ", "amarelo" );
    cout << usuarioMaisAcertou << " ( " << maiorAcerto << " )" << endl;
    textoColorido( "\nUsuario que mais errou..: ", "amarelo" );
    cout << usuarioMaisErrou << " ( " << maiorErro << " )" << endl;
    cout << endl;
    replica( '-', REPETE + 10 );
    pausa();
  } else
    mensagem( "RESULTADOS NAO ENCONTRADOS" );
}


simulado.cpp
/*
 Programa.......: Simulado Virtual
 Projetado em...: 26 de Outubro  de 2006
 Criado em......: 02 de Novembro de 2006
 Desenvolvedores: Luiz Sanches
 Objetivo.......: Atraves da leitura de arquivos textos, criar rapidamente
                  simulados virtuais para qualquer instituicao de ensino
                  obtendo um resultado rapido dos testes realizados pelos alunos.
*/

// Inclusao de bibliotecas
#include <cstdlib>
#include <cstring>
#include <iostream>
#include "funcoes.h"
#include "arquivos.h"
#include "resultados.h"
#include "constantes.h"

// Prototipos de funcoes
void realizaSimulados( void );
void criaArquivos( void );
void limpaTela( void );
void pausa( void );
void replica( char, int );
void mensagem( const char * const );
void titulo( const char * const , char, int );
void corFonte( const char * const, const char * const );
void textoColorido( const char * const, const char * const );
void sobreSistema( void );
void resultados( void );

char nomeUsuario[15]; // Variavel global para o nome do usuario

// Menu principal
int main() {
	criaArquivos(); // Cria os arquivos de dados que nao existem

	// Pede a digitacao do nome do usuario
	titulo( "Simulado Virtual - Acesso ao Sistema", '=', REPETE + 10 );
	textoColorido( "\nDigite seu primeiro nome: ", "amarelo" );
	corFonte( "verde", COR_FUNDO );
	cin >> nomeUsuario;
	corFonte( "branco", COR_FUNDO );
	replica( '-', REPETE + 10 );

	int opcao = 1;

	while ( opcao != 0 ) {
		titulo( "Simulado Virtual", '=', REPETE );
		textoColorido( "Por: Luiz Sanches\n", "verde" );

		corFonte( "branco", COR_FUNDO );
		replica( '-', REPETE );
		cout << "\n1 - Realizar Simulados\n\n";
		cout << "2 - Resultados\n\n";
		cout << "3 - Sobre o Sistema\n\n";
		textoColorido( "0 - Sair\n\n", "amarelo" );
		replica( '-', REPETE );

		textoColorido( "Escolha uma opcao: ", "amarelo" );
		corFonte( "verde", COR_FUNDO );
		cin >> opcao;
		corFonte( "branco", COR_FUNDO );

		switch ( opcao ) {
			case 0:
				break;
			case 1:
				realizaSimulados();
				break;
			case 2:
				resultados();
				break;
			case 3:
				sobreSistema();
				break;
			default:
				mensagem( "OPCAO INVALIDA" );
				break;
		}
	}

	corFonte( "branco", "preto" );
	limpaTela();
	return 0;
}

// Rotina para realizar os simulados
void realizaSimulados( void ) {
	// Criacao de constantes
	const int MAX = 80;

	// Criacao de variaveis
	int op = 1;
	char opSimulado;
	char opPergunta;
	char opResposta;
	char respostaCorreta;
	int respostaMaxima;

	int i;

	int achouSimulado;
	int achouResposta;

	float numeroQuestoes;
	float numeroAcertos;
	float numeroErros;

	float percentualTotal = 0;
	float percentualAcertos = 0;
	float percentualErros = 0;

	time_t dataSistema;
	struct tm *dataExpandida;
	char dataAtual[11];

	// Criacao de vetores
	char texto[ MAX ] = { "" };
	char textoPergunta[ MAX ] = { "" };
	char simulado[ MAX ] = { "" };

	while ( op != 0 ) {
		achouSimulado = 0;
		achouResposta = 0;

		numeroQuestoes = 0;
		numeroAcertos = 0;
		numeroErros = 0;

		limpaTela();

		// Abertura do arquivo de simulados
		ifstream inSimulados( "simulados.dat", ios::in );

		// Se nao conseguir abrir o arquivo, emite uma mensagem de erro
		if ( !inSimulados ) {
			cerr << "\nArquivo nao pode ser aberto\n";
			inSimulados.close();
			pausa();
		}
		else {
			// Se conseguiu abrir o arquivo, exibe o conteudo do mesmo
			titulo( "Realizar Simulados", '=', REPETE );

			// Exibe os simulados
			while ( inSimulados ) {
				inSimulados.getline( texto, MAX );

				if ( strlen(texto) > 0 ) {
					cout << '\n' << texto[ 0 ] << " - "; // Imprime o codigo do simulado

					// Imprime o nome do simulado
					for ( i = 2; texto[ i ] != '\0'; i++ )
						cout << texto[ i ];

					cout << '\n';
				}
			}

			textoColorido( "\n0 - Voltar\n\n" , "amarelo" );
			replica( '-', REPETE );

			// Pede a digitacao do numero do simulado
			textoColorido( "Escolha o simulado: ", "amarelo" );
			corFonte( "verde", COR_FUNDO );
			cin >> op;
			corFonte( "branco", COR_FUNDO );

			if ( op == 0 )
				break;

			// Conversao de inteiro para caractere
			op = op + 48; // Em ascii 48-57 = 0-9
			opSimulado = (char) op;

			inSimulados.clear(); // Desliga fim de arquivo para proxima entrada
			inSimulados.seekg( 0 ); // Vai para o inicio do arquivo

			// Procura o codigo do simulado no arquivo simulados.dat
			while ( inSimulados ) {
				inSimulados.getline( texto, MAX ); // Seleciona a linha do arquivo

				// Se a posicao zero da linha for igual ao simulado escolhido
				if ( texto[0] == opSimulado ) {
					for ( i = 0; i < MAX; i++ )
						simulado[ i ] = texto[ i ];

					achouSimulado = 1;
					break;
				}
			}
			inSimulados.close(); // Fecha o arquivo de simulados

			if ( achouSimulado ) { // Abre o arquivo perguntas.dat
				ifstream inPerguntas( "perguntas.dat", ios::in );

				if ( !inPerguntas ) {
					cerr << "\nArquivo de perguntas nao pode ser aberto\n";
					inPerguntas.close();
					pausa();
				} else {
					// Procura as perguntas no arquivo
					while ( inPerguntas ) {
						inPerguntas.getline( textoPergunta, MAX );

						// Se a linha for do simulado selecionado, realiza a pergunta
						if ( textoPergunta[0] == opSimulado ) {
							numeroQuestoes++;

							op = 1;
							// Executa o laco enquanto nao abandonar o simulado
							while ( op != 0 ) {
								limpaTela();

								replica( '=', REPETE + 30 );
								corFonte ( "amarelo", COR_FUNDO );

								// Exibe o nome do simulado
								for ( i = 2; i < strlen(simulado); i++ )
									cout << simulado[ i ];

								textoColorido(" - Usuario: " , "verde" );
								textoColorido(nomeUsuario , "verde" );
								cout << '\n';
								corFonte ( "branco", COR_FUNDO );
								replica( '=', REPETE + 30 );
								cout << '\n';

								// Exibe a pergunta
								corFonte( "verde", COR_FUNDO );

								opPergunta = textoPergunta[ 2 ];

								//cout << opPergunta << " - ";

								for ( i = 4; textoPergunta[ i ] != '\0'; i++ )
									cout << textoPergunta[ i ];

								cout << "\n\n";

								corFonte( "branco", COR_FUNDO );

								// Abre o arquivo de respostas
								ifstream inRespostas( "respostas.dat", ios::in );

								if ( !inRespostas ) {
									cerr << "\nArquivo de respostas nao pode ser aberto\n";
									inRespostas.close();
									pausa();
								} else {
									// Procura as respostas da pergunta no arquivo
									respostaCorreta = ' ';
									respostaMaxima = 0;
									achouResposta = 0;

									while ( inRespostas ) {
										inRespostas.getline( texto, MAX );

										// Se a linha for do simulado
										// e pergunta selecionada, exibe a resposta
										if ( texto[ 0 ] == opSimulado && texto[ 2 ] == opPergunta ) {
											achouResposta = 1;
											respostaMaxima++;

											if ( texto[ 6 ] == 's' ) // Se a resposta for correta='s'
												respostaCorreta = texto[ 4 ]; // Pega o numero da resposta

											cout << texto[ 4 ] << " - "; // Imprime o numero da resposta

											// Exibe a resposta
											for ( i = 8; texto[ i ] != '\0'; i++ )
												cout << texto[ i ];

											cout << endl << endl;
										}
									}
									inRespostas.close();

									// Se achou a resposta da pergunta
									if ( achouResposta ) {
										textoColorido( "0 - Abandona\n\n", "amarelo" );
										replica( '-', REPETE + 30 );
										textoColorido( "Escolha a resposta: ", "amarelo" );
										corFonte( "verde", COR_FUNDO );
										cin >> op;
										corFonte( "branco", COR_FUNDO );

										if ( op == 0 ) // Se for zero abandona o simulado
											break;

										// Se a opcao escolhida for valida
										if ( op >= 1 && op <= respostaMaxima ) {
											// Conversao de inteiro para caractere
											op = op + 48; // Em ascii 48-57 = 0-9
											opResposta = (char) op;

											if ( opResposta == respostaCorreta ) {
												numeroAcertos++;
												mensagem( "PARABENS! RESPOSTA CORRETA" );
												break;
											} else {
												numeroErros++;
												mensagem( "QUE PENA, RESPOSTA INCORRETA" );
												break;
											}
										} else
											mensagem( "RESPOSTA INVALIDA" );

									} // Fim da condicao se achou a resposta
								} // Fim da condicao se abriu o arquivo de respostas
							} // Fim do laco de escolha da resposta

							if ( op == 0 ) { // Se for zero abandona o simulado
								numeroQuestoes--; // Substrai o n. de questoes para o resultado
								break;
							}

						} // Fim da condicao se encontrou a pergunta do simulado escolhido
					} // Fim do laco de leitura do arquivo de perguntas
					inPerguntas.close(); // Fecha o arquivo de perguntas

					if ( numeroQuestoes > 0 ) {
						// Exibe o resultado do simulado
						percentualAcertos = ( numeroAcertos / numeroQuestoes ) * 100;
						percentualErros = ( numeroErros / numeroQuestoes ) * 100;
						percentualTotal = percentualAcertos + percentualErros;

						titulo( "Resultado do Simulado", '=', REPETE );
						textoColorido( "\nAcertos: ", "amarelo" );
						cout << numeroAcertos << " - " << (int) percentualAcertos << "%\n";

						textoColorido( "\nErros..: ", "amarelo" );
						cout << numeroErros << " - " << (int) percentualErros << "%\n\n";

						replica( '-', REPETE );
						textoColorido( "\nTotal..: ", "amarelo" );
						cout << numeroQuestoes << " - " << percentualTotal << "%\n";

						// Recupera a data atual do sistema
						dataSistema = time( NULL );
						dataExpandida = localtime( &dataSistema );
						strftime( dataAtual, 11, "%d %m %Y", dataExpandida );

						// Grava no arquivo resultados.dat
						ofstream outResultados( "resultados.dat", ios::app );

						outResultados << dataAtual << ' ' <<nomeUsuario << ' ' << opSimulado
									<< ' ' << numeroAcertos << ' ' << numeroErros
									<< ' ' << numeroQuestoes << '\n';
						outResultados.close();

						pausa();
					}
				} // Fim da condicao se abriu o arquivo de perguntas
			} else {
				mensagem( "O SIMULADO NAO FOI ENCONTRADO" );

			} // Fim da condicao se achou o simulado no arquivo
		} // Fim da condicao se abriu o arquivo de simulados
	} // Fim do laco de escolha do simulado
} // Fim da rotina para realizar simulados

// Procedimento para exibir a tela de informacao sobre o sistema
void sobreSistema( void ) {
	titulo( "Sobre o Simulado Virtual", '=', REPETE + 18 );
	cout << "\n Instituto de Estudos Superiores da Amazonia - IESAM";
	cout << "\n Curso de Bacharelado em Sistemas de Informacao";
	cout << "\n 1o Ano - Novembro de 2006";
	cout << "\n Algoritmo e Programacao - C++";
	cout << "\n Profa. Lenilda Pinheiro";
	cout << "\n Aluno: Luiz Sanches\n\n";
	replica( '-', REPETE + 18 );
	pausa();
}

